packages/dts-loader/src/index.ts

```diff
@@ -155,11 +155,27 @@ interface LoaderOptions {
   typesOutputDir: string
 }
 
+function hash(content: string) {
+  const hash = crypto.createHash("md5");
+  hash.update(content);
+  return hash.digest('hex');
+}
+
 function makeLoader(
   context: LoaderContext<Partial<LoaderOptions>>,
   loaderOptions: LoaderOptions,
   content: string
 ) {
+  const fileName = context.resourcePath;
+  const hashedContent = hash(content);
+  if (files[fileName] && files[fileName].hashedContent !== hashedContent) {
```

Comment:
The changes look good to me, but there is one limitation: it won't emit `.d.ts` files for newly added files. The reason is TS language server was created with `getScriptFileNames: () => rootFileNames`. We probably can create new language server when the file hasn't been added.